gateway-shell/src/main/java/org/apache/knox/gateway/shell/KnoxShellTable.java

```diff
@@ -654,5 +655,146 @@ public KnoxShellTable regex(String regex) {
       }
       return table;
     }
+
+    public KnoxShellTable filter(Predicate p) {
+        KnoxShellTable table = new KnoxShellTable();
+        table.headers.addAll(headers);
+        for (List<String> row : rows) {
+          if (p.test(row.get(index))) {
+            table.row();
```

Comment:
What does this line do without any arguments? Seems like it isn't necessary?

Reply:
If it is necessary, would be good to add a comment saying what it is for.

Reply:
I'll add a comment to clarify that.

Reply:
Where does `index` come from?

Reply:
index is a member variable in KnoxShellTableFilter